### PR TITLE
Re-implement tool that parses and reports on a trip csv file

### PR DESCRIPTION
**Pull request by mariusfeteanu**
### What and why
The original script was hard to read and test, so I kind of re-wrote everything. The only bug in the original test was that it didn't skip headers when loading data.

### Feedback
Feedback is a gift, so a few points on the original code before going on to the changes:
- Too many short variable names, this is fine in local scope (one function) but they still have to mean something. `a` for argument, `j` for journey are sometimes fine if the scope fits in one screen (for example in an anonymous function). `x` for the first argument and `y` for the file content is much harder to remember.
- `while` loops can be equivalent to functional operators (map, foreach) and are mostly fine. However if you already have a list in hand you might as well use it as using `while` will make people read the code twice to see if you have a good reason.
- When parsing user input (`var x = args(0)`) it's good to fail explicitly with an error they can understand. In other cases it's fine to fatal hard and fast so don't try to write a friendly error message for everything.
- Fine to leave TODOs in code but it's a good idea to mark them as such with a TODO keyword or something; this is so that people can search for them and solve them.
- Try not to commit broken tests, better to skip them and leave a TODO if no time.

### Project changes
- added `.gitignore`
- added a launcher so you can run the app like this: `mvn scala:run -DaddArgs="src/test/resources/2021-10-05_journeys.csv"`
- added a formatter config file for vscode

### Code changes:
- Moved all data structures to the `model` package. Nothing fancy just a `Journey` and `JourneyStats` case classes
- Moved all transformations to the `transform` package. These have to be stateless so they are easy to test
- Created a very simple IO class in the `io` package so we can test the dirty business of reading the file separately. A lot of TODOs still in that file
- The `ProcessDataFile` entry point is relatively straightforward now
- As can be seen from the above the main idea here is to separate things that store state from those that don't so we can test them separately
- The `t(journeys)` calls from the main script are still not very readable, but I couldn't find a prettier way

### Tests
- The examples are really verbose, but I still thought it better to have them with the tests. I just collapse the sections in my editor
- The entry point tests are rather basic, assuming the important stuff was tested for the other modules
- also testing a variety of files (empty files and such)
- No code coverage

### CODE REVIEW

**Review by BasilBibi — APPROVED** [review]
Good abstractions (LatLong?): 8/10
Nice, well named atomic functions with very few params: 10/10
Uses scala collections and inbuilt manipulations : 10/10
Good variable names although I did not like {JourneysTransform => t} : 10/10
Used drop : 2/2
Filter on read : 5/5
Journeys > 90 mins : 5/5
Average Speed : 5/5
Tot mileage by driver : 5/5  Nice clear solution - I like it better than mine
Most active driver : 5/5
65%

Bonus for additional tests and data files : 15%
Bonus for reorganising repo layout : 10%

90% overall

===================
Did not remove duplicates
Output unsorted
Added unspecified "kilometers" in output

File: src/test/scala/ai/humn/telematics/transform/JourneysTransformTest.scala [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+          endLat = 0.0,
+          endLon = 0.0,
+          startOdometer = 0,
+          endOdometer = 100
+        )
+      )
+    );
+    assert(js.removeInvalid().isEmpty);
+  }
+  it must "ignore a negative or zero odometer journey" in {
```
Comment (woj-i) [inline src/test/scala/ai/humn/telematics/transform/JourneysTransformTest.scala:34]: nice!

**Review by woj-i — APPROVED** [review]
As Basil mentioned it's a solid work. I tried to suggest possible improvements.

**Review by florin1288 — APPROVED** [review]
LGTM

**Review by a-romero — APPROVED** [review]
looks really good
